server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessageFactoryTest.java

```diff
@@ -338,6 +341,55 @@ public void invalidAddressesShouldBeAllowed() throws Exception {
     }
 
     @Test
+    public void dateFromHeaderShouldBeUsedIfPresent() throws Exception {
+        String headers = "From: user <userdomain>\n"
+            + "To: user1 <user1domain>, user2 <user2domain>\n"
+            + "Cc: usercc <userccdomain>\n"
+            + "Bcc: userbcc <userbccdomain>\n"
+            + "Date: Wed, 17 May 2017 14:18:52 +0200\n"
+            + "Subject: test subject\n";
+
+        MetaDataWithContent testMail = MetaDataWithContent.builder()
+            .uid(MessageUid.of(2))
+            .flags(new Flags(Flag.SEEN))
+            .size(headers.length())
+            .internalDate(INTERNAL_DATE)
+            .content(new ByteArrayInputStream(headers.getBytes(Charsets.UTF_8)))
+            .attachments(ImmutableList.of())
+            .mailboxId(MAILBOX_ID)
+            .messageId(new TestMessageId.Factory().generate())
+            .build();
+
+        Message testee = messageFactory.fromMetaDataWithContent(testMail);
+
+        assertThat(testee.getDate()).isEqualTo(ZonedDateTime.of(2017, 05, 17, 14, 18, 52, 00, ZoneId.of("Europe/Paris")));
+    }
+
+    @Test
+    public void internalDateShouldBeUsedIfNoDateInHeaders() throws Exception {
+        String headers = "From: user <userdomain>\n"
+            + "To: user1 <user1domain>, user2 <user2domain>\n"
+            + "Cc: usercc <userccdomain>\n"
+            + "Bcc: userbcc <userbccdomain>\n"
+            + "Subject: test subject\n";
+
+        MetaDataWithContent testMail = MetaDataWithContent.builder()
+            .uid(MessageUid.of(2))
+            .flags(new Flags(Flag.SEEN))
+            .size(headers.length())
+            .internalDate(INTERNAL_DATE)
+            .content(new ByteArrayInputStream(headers.getBytes(Charsets.UTF_8)))
+            .attachments(ImmutableList.of())
+            .mailboxId(MAILBOX_ID)
+            .messageId(new TestMessageId.Factory().generate())
+            .build();
+
+        Message testee = messageFactory.fromMetaDataWithContent(testMail);
+
+        assertThat(testee.getDate()).isEqualTo(ZONED_DATE);
+    }
+
```

Comment:
We had a long conversation on Mattermost about that. To sum-up we agreed that Instant was the correct type for what we wanted.

The idea is that with `ZonedTimeZone` a new developer (me for example two day ago) must know that :
 - We use `ZonedDatetime` but we are only allow to use `ZonedDatetime` with UTC time-zone otherwise when serializing the time-zone appears in the date when JMAP spec specify date should be in UTC only.
 - The jmap spec ask you to only use UTC timezone
 - We use `ZonedDatetime` but we are loosing any time-zone information from the email date header.

Using Instant solve all of those issue and moreover it simplify the code